cmake/Jemalloc.cmake

```diff
@@ -0,0 +1,40 @@
+add_library(jemalloc INTERFACE)
+if(NOT WIN32)
+  # We don't want to use jemalloc on Windows
+  find_path(JEMALLOC_INCLUDE_DIR
+    NAMES
+      jemalloc/jemalloc.h
+    PATH_SUFFIXES
+      include
+    )
+  find_library(JEMALLOC NAMES libjemalloc.a)
+  find_library(JEMALLOC_PIC NAMES libjemalloc_pic.a)
+  add_library(im_jemalloc_pic STATIC IMPORTED)
+  add_library(im_jemalloc STATIC IMPORTED)
+  if(JEMALLOC_INCLUDE_DIR AND JEMALLOC AND JEMALLOC_PIC)
+    set_target_properties(im_jemalloc_pic PROPERTIES IMPORTED_LOCATION "${JEMALLOC_PIC}")
+    set_target_properties(im_jemalloc PROPERTIES IMPORTED_LOCATION "${JEMALLOC}")
+    target_include_directories(jemalloc PUBLIC "${JEMALLOC_INCLUDE_DIR}")
```

Comment:
Should these be INTERFACE instead of PUBLIC, since jemalloc is an INTERFACE library?

Reply:
you're right, I changed this to `INTERFACE`